### PR TITLE
HOCS-3644 - Record Internal Review and Appeal events

### PR DESCRIPTION
**Pull request by agraham19**
* Add appeals to Actions bar in FOI

### CODE REVIEW

File: server/services/forms/index.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 const formRepository = require('./schemas/index');
 const formDecorator = require('./schemas/decorators/form-decorator');
+
 const {
     ADD_TEMPLATE, ADD_STANDARD_LINE, IS_MEMBER, ADD_MEMBER, SELECT_MEMBER, ADD_CORRESPONDENT, UPDATE_CORRESPONDENT,
     REMOVE_CORRESPONDENT, ADD_TOPIC, REMOVE_TOPIC, CREATE_CASE, CREATE_AND_ALLOCATE_CASE, BULK_CREATE_CASE,
     ADD_DOCUMENT, REMOVE_DOCUMENT, MANAGE_DOCUMENTS, MANAGE_PEOPLE, ADD_CONTRIBUTION, ADD_ADDITIONAL_CONTRIBUTION,
-    EDIT_CONTRIBUTION, APPLY_CASE_DEADLINE_EXTENSION, CONFIRMATION_SUMMARY, ADD_APPROVAL_REQUEST, EDIT_APPROVAL_REQUEST
+    EDIT_CONTRIBUTION, APPLY_CASE_DEADLINE_EXTENSION, CONFIRMATION_SUMMARY, ADD_CASE_APPEAL, EDIT_CASE_APPEAL, MANAGE_CASE_APPEALS: MANAGE_CASE_APPEALS,
```
Comment (gary-newbolt) [inline server/services/forms/index.js:8]: You have `MANAGE_CASE_APPEALS: MANAGE_CASE_APPEALS` here... is that a mistake? Should it be here just once?

File: server/services/forms/schemas/manage-appeals-foi.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const somuType = await listServiceInstance.getFromStaticList(
+        'SOMU_TYPES',
+        ['FOI', 'APPEAL']
+    );
+
+    const choices = await listServiceInstance.fetch(
+        'FOI_APPEAL_TYPES',
+    );
+
+    console.log('CHOICES: ' + JSON.stringify(choices));
```
Comment (gary-newbolt) [inline server/services/forms/schemas/manage-appeals-foi.js:19]: lets get this removed please

File: server/services/forms/schemas/update-appeal-foi.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                ])
+                .withProp('visibilityConditions', [
+                    {
+                        'conditionPropertyName': 'appealStatus',
+                        'conditionPropertyValue': 'appealComplete'
+                    }
+                ])
+                .build()
+        ).withField(
+            Component('radio', 'complex')
```
Comment (gary-newbolt) [inline server/services/forms/schemas/update-appeal-foi.js:87]: I've made some comments about the naming on this in the `hocs-data` repo changes.

File: server/services/forms/schemas/update-appeal-foi.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                ])
+                .withProp('visibilityConditions', [
+                    {
+                        'conditionPropertyName': 'appealStatus',
+                        'conditionPropertyValue': 'appealComplete'
+                    }
+                ])
+                .build()
+        ).withField(
+            Component('text-area', 'notes')
```
Comment (gary-newbolt) [inline server/services/forms/schemas/update-appeal-foi.js:102]: like above, I've made a comment on the naming in the `hocs-data` repo.

File: src/shared/common/forms/composite/somu-list.jsx [hunk 5]
```diff
@@ -105,6 +105,31 @@ class SomuTableRenderer {
         </>);
     }
 
+    renderFoiAppeal(somuItem) {
```
Comment (gary-newbolt) [inline src/shared/common/forms/composite/somu-list.jsx:108]: The method doesn't seem to be geared to FOI other than we're the first to use it... can it be chanced easily to just `renderAppeal`?

File: src/shared/common/forms/composite/somu-list.jsx [hunk 6]
```diff
@@ -105,6 +105,31 @@ class SomuTableRenderer {
...
+                    title={somuItem.appealStatus === 'appealComplete' ? 'Complete': 'Pending'}>
+                    {somuItem.appealStatus === 'appealComplete' ? 'Complete': 'Pending'}
+                </label>
+            </td>
+        </>
+        );
+    }
+
+
+    renderStatusColumn(status, { contributionDueDate }) {
```
Comment (gary-newbolt) [inline src/shared/common/forms/composite/somu-list.jsx:132]: Whoops... what's happened here? Merge/Rebase error?

File: server/services/forms/index.js (hunk 1 above)
Comment (agraham19) [reply server/services/forms/index.js]: I have addressed this

File: server/services/forms/schemas/manage-appeals-foi.js (hunk 2 above)
Comment (agraham19) [reply server/services/forms/schemas/manage-appeals-foi.js]: I have addressed this

File: server/services/forms/schemas/update-appeal-foi.js (hunk 3 above)
Comment (agraham19) [reply server/services/forms/schemas/update-appeal-foi.js]: I have addressed this

File: server/services/forms/schemas/update-appeal-foi.js (hunk 4 above)
Comment (agraham19) [reply server/services/forms/schemas/update-appeal-foi.js]: I have addressed this

File: src/shared/common/forms/composite/somu-list.jsx (hunk 5 above)
Comment (agraham19) [reply src/shared/common/forms/composite/somu-list.jsx]: I have addressed this

File: src/shared/common/forms/composite/somu-list.jsx (hunk 6 above)
Comment (agraham19) [reply src/shared/common/forms/composite/somu-list.jsx]: I have addressed this

**Review by gary-newbolt — APPROVED** [review]
great, thanks

**Pull request merged**